### ISSUE TITLE
Support Precise Literals from Bitwise Operators

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#8419]
Currently, for bitwise operations that can be statically determined, an exact literal isn't narrowed:

```python
from typing import Final, reveal_type

MY_CONST: Final = 1 | 2

reveal_type(MY_CONST) # Revealed type is `Type of "MY_CONST" is "int"`
```

The same goes for the other bitwise operators (`>>`, `<<`, `^`, etc).

[MyPy does support exact literals in these cases](https://mypy-play.net/?mypy=latest&python=3.12&gist=27f99402249cb8a3b139af148234c350), so parity could be nice for stub/library maintainers. I think the most useful being bit-shifting literals.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by max-muoto** [comment]
> This will be included in the next release.

Thanks Eric!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.372.
